planner/core/exhaust_physical_plans.go

```diff
@@ -2328,7 +2330,11 @@ func (la *LogicalAggregation) getHashAggs(prop *property.PhysicalProperty) []Phy
 	if la.HasDistinct() {
 		// TODO: remove AllowDistinctAggPushDown after the cost estimation of distinct pushdown is implemented.
 		// If AllowDistinctAggPushDown is set to true, we should not consider RootTask.
-		if !la.ctx.GetSessionVars().AllowDistinctAggPushDown {
+		considerRootTask := !la.ctx.GetSessionVars().AllowDistinctAggPushDown
```

Comment:
This comments and variable names are confusing: considerRootTask actually means "only consider root" instead of consider all kinds of paths. 

Indeed it's ok to put these conditions into one condition with "&&". no need to write multiple lines.

Reply:
updated